metaflow/extension_support.py

```diff
@@ -0,0 +1,441 @@
+                                _extension_points[idx]
+                            ].get(dist.metadata["Name"], None)
+                            if cur_pkg is not None:
+                                if (
+                                    config_module is not None
+                                    and cur_pkg.config_module is not None
+                                ):
+                                    raise RuntimeError(
+                                        "Package '%s' defines more than one "
+                                        "configuration file for '%s': '%s' and '%s'"
+                                        % (
+                                            dist.metadata["Name"],
+                                            _extension_points[idx],
+                                            config_module,
+                                            cur_pkg.config_module,
+                                        )
+                                    )
+                                if config_module is not None:
+                                    _ext_debug(
+                                        "\tFound config file '%s'" % config_module
+                                    )
+                                    extension_points_to_pkg[_extension_points[idx]][
+                                        dist.metadata["Name"]
+                                    ] = MFExtPackage(
+                                        package_name=dist.metadata["Name"],
+                                        tl_package=parts[1],
+                                        config_module=config_module,
+                                    )
+                            else:
+                                _ext_debug(
+                                    "\tExtends '%s' with config '%s'"
+                                    % (_extension_points[idx], config_module)
+                                )
+                                extension_points_to_pkg[_extension_points[idx]][
+                                    dist.metadata["Name"]
+                                ] = MFExtPackage(
+                                    package_name=dist.metadata["Name"],
+                                    tl_package=parts[1],
+                                    config_module=config_module,
+                                )
+                            break
+    extension_points_to_pkg.default_factory = list
+
+    # At this point, we have all the packages that contribute to EXT_PKG,
+    # we now check to see if there is an order to respect based on dependencies. We will
+    # return an ordered list that respects that order and is ordered alphabetically in
+    # case of ties.
+    pkg_to_reqs_count = {}
+    req_to_dep = {}
+    mf_ext_packages_set = set(mf_ext_packages)
+    for pkg_name in mf_ext_packages:
+        req_count = 0
+        req_pkgs = [x.split()[0] for x in metadata.requires(pkg_name)]
```

Comment:
`metadata.requires(pkg_name)` returns a `Distribution(pkg_name).requires` which can be None. Can we have : 
```python
distmd = metadata.requires(pkg_name)
if distmd:
    req_pkgs = [x.split()[0] for x in metadata.requires(pkg_name)]
    ....
    ...
```

Reply:
Good catch. Will fix.